app/lib/synthetic_study_populator.rb

```diff
@@ -0,0 +1,61 @@
+
+# class to populate synthetic studies from files.
+# See db/seed/synthetic_studies for examples of the file formats
+class SyntheticStudyPopulator
+  DEFAULT_SYNTHETIC_STUDY_PATH = Rails.root.join('db', 'seed', 'synthetic_studies')
+  # populates all studies defined in db/seed/synthetic_studies
+  def self.populate_all(user: User.first)
+    study_names = Dir.glob(DEFAULT_SYNTHETIC_STUDY_PATH.join('*')).select {|f| File.directory? f}
+    study_names.each do |study_name|
+      populate(study_name, user: user)
+    end
+  end
+
+  # populates the synthetic study specified in the given folder (e.g. ./db/seed/synthetic_studies/blood)
+  # destroys any existing studies and workspace data corresponding to that study
+  def self.populate(synthetic_study_folder, user: User.first)
+    if (synthetic_study_folder.exclude?('/'))
+      synthetic_study_folder = DEFAULT_SYNTHETIC_STUDY_PATH.join(synthetic_study_folder).to_s
+    end
+    study_info_file = File.read(synthetic_study_folder + '/study_info.json')
+    study_config = JSON.parse(study_info_file)
+
+    puts("Populating synthetic study from #{synthetic_study_folder}")
+    study = create_study(synthetic_study_folder, study_config, user)
+    add_files(study, study_config, synthetic_study_folder, user)
+  end
+
+  private
+
+  def self.create_study(synthetic_study_folder, study_config, user)
+    user_suffix = '-' + (`git config user.email`.strip[0, 2] || 'xx')
```

Comment:
Personally, whenever I see backticks for the `system` command, I worry that this is the kind of thing that can easily break or lead to unintended return values.  Also, since we have a `user` object already here, we can strip off the username before `@` easily enough to get the same value.

Reply:
agree on this being hacky, the idea was it would be short-lived while we work out the separate workspace issues.  But since 'short-lived' is always a goal rather than guarantee, I'll pull from the user like you suggest :)